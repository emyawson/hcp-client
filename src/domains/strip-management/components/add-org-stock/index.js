export * from './add-org-stock.component';
