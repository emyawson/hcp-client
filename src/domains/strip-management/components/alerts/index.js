export * from './alerts-header';
export * from './alerts-threshold-row';
export * from './alerts-threshold';
export * from './alerts-counter';
