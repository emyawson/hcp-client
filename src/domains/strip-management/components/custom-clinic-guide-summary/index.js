export * from './custom-clinic-guide-summary.component';
