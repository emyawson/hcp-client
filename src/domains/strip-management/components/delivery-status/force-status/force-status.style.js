import styled from 'styled-components';

import { spacing } from 'src/core';
import { InputTextArea } from 'src/components/input/input.style';

export const ForceStatusRowDiv = styled.div`
  display: flex;
  flex-wrap: nowrap;
  padding: ${spacing.three} 0;
  width: 100%;
`;

export const ForceStatusControlsDiv = styled.div`
  display: flex;
  flex-wrap: nowrap;
  flex: 1 0 20%;
  justify-content: center;
  padding-right: ${spacing.three};
  min-width: 8rem;
`;

export const ForceStatusControlSpan = styled.span`
  margin: 0 ${spacing.two};
`;

export const ForceStatusSubmitDiv = styled.div`
  align-items: center;
  display: flex;
  justify-content: flex-end;
`;

export const ForceStatusReasonTextArea = InputTextArea.extend`
  flex: 1 1 80%;
`;
