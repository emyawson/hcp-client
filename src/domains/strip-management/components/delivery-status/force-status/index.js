export * from './force-status.component';
