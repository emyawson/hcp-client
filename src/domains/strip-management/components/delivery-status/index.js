export * from './delivery-status-icon';
export * from './force-status';
export * from './force-status-button';
export * from './strip-status-details';
export * from './traffic-light';
