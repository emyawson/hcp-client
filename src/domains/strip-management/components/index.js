export * from './add-org-stock';
export * from './alerts';
export * from './delivery-status';
export * from './patient-strip-stock';
export * from './prescriptions';
export * from './strip-delivery-alerts';
export * from './home-delivery-card';
export * from './total-stock-card';
export * from './org-stock-history';
export * from './quantity-slider';
export * from './history-header';
export * from './custom-clinic-guide-summary';
export * from './labelled-grid-item';
export * from './custom-guide-history';
