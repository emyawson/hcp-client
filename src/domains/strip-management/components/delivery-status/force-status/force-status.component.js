import React from 'react';
import { equals, filter, isNil, not, pipe } from 'ramda';

import { TRAFFIC_LIGHT_STATES, STATUS_COMMENT_MAX_LENGTH } from 'src/core';
import { Button, ControlText, ControlButton, Form } from 'src/components';
import { translate } from 'src/i18n';

import {
  ForceStatusSubmitDiv,
  ForceStatusControlsDiv,
  ForceStatusControlSpan,
  ForceStatusReasonTextArea,
  ForceStatusRowDiv,
} from './force-status.style';

import { DeliveryStatusIcon, ForceStatusButton } from '../../delivery-status';

const { DELIVER, DO_NOT_DELIVER, DISABLED } = TRAFFIC_LIGHT_STATES;

const forcibleTrafficStates = [DELIVER, DO_NOT_DELIVER];

export const ForceStatus = ({
  currentStatus = DISABLED,
  disabled = false,
  forceTrafficStatus,
  modelPath,
  setDeliveryStatusRequest,
  patientId,
}) => {
  const submitForcedStatus = ({ status, comment }) => {
    setDeliveryStatusRequest({ patientId, status, comment });
  };
  const isNotCurrentStatus = pipe(
    equals(currentStatus),
    not,
  );
  const forceStatusControls = filter(isNotCurrentStatus, forcibleTrafficStates);
  return (
    <Form model={modelPath} onSubmit={submitForcedStatus}>
      <ForceStatusRowDiv>
        <ForceStatusControlsDiv>
          {forceStatusControls.map(trafficStatus => (
            <ForceStatusControlSpan
              key={`ForceStatusButtonContainer-${trafficStatus}`}
            >
              <ForceStatusButton
                key={`ForceStatusButton-${trafficStatus}`}
                disabled={disabled || equals(trafficStatus, forceTrafficStatus)}
                modelPath=".status"
                status={trafficStatus}
              />
            </ForceStatusControlSpan>
          ))}
        </ForceStatusControlsDiv>
        <ControlText
          model=".comment"
          component={ForceStatusReasonTextArea}
          placeholder={translate('stripDelivery.forceStatus.comment')}
          maxLength={STATUS_COMMENT_MAX_LENGTH}
        />
      </ForceStatusRowDiv>
      <ForceStatusSubmitDiv>
        <DeliveryStatusIcon status={forceTrafficStatus} noIcon={true} />
        <div>
          <ControlButton
            component={Button}
            disabled={disabled || isNil(forceTrafficStatus)}
            model="stripDelivery.forceTrafficStatus"
            type="submit"
            label={translate('stripDelivery.forceStatus.submit')}
          />
        </div>
      </ForceStatusSubmitDiv>
    </Form>
  );
};
