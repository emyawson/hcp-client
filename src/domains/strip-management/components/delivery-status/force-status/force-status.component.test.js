import React from 'react';
import { shallow } from 'enzyme';

import { TRAFFIC_LIGHT_STATES } from 'src/core';
import { Form } from 'src/components';

import { ForceStatus } from './force-status.component';

describe('Force status component', () => {
  const mockBaseProps = {
    forceTrafficStatus: TRAFFIC_LIGHT_STATES.DELIVER,
    modelPath: 'testModel',
    setDeliveryStatusRequest: () => {},
    patientId: '123',
  };
  it('Renders correctly', () => {
    const mockProps = {
      ...mockBaseProps,
      currentStatus: TRAFFIC_LIGHT_STATES.DO_NOT_DELIVER,
      disabled: true,
    };
    const wrapper = shallow(<ForceStatus {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Renders with default props', () => {
    const wrapper = shallow(<ForceStatus {...mockBaseProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Calls request action when form is submitted', () => {
    const mockRequestAction = jest.fn().mockName('set delivery status request');
    const mockProps = {
      ...mockBaseProps,
      setDeliveryStatusRequest: mockRequestAction,
    };
    const wrapper = shallow(<ForceStatus {...mockProps} />);
    wrapper
      .find(Form)
      .simulate('submit', { status: TRAFFIC_LIGHT_STATES.DELIVER });

    expect(mockRequestAction).toBeCalledWith({
      patientId: '123',
      status: TRAFFIC_LIGHT_STATES.DELIVER,
    });
  });
});
