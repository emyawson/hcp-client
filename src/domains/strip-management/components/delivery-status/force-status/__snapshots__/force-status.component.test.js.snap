// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Force status component Renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentStatus="DO_NOT_DELIVER"
    disabled={true}
    forceTrafficStatus="DELIVER"
    modelPath="testModel"
    patientId="123"
    setDeliveryStatusRequest={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.div>
            <styled.span>
              <Unknown
                disabled={true}
                modelPath=".status"
                status="DELIVER"
              />
            </styled.span>
          </styled.div>
          <Unknown
            component={[Function]}
            maxLength={255}
            model=".comment"
            placeholder="Force status comments"
          />
        </styled.div>,
        <styled.div>
          <Unknown
            noIcon={true}
            status="DELIVER"
          />
          <div>
            <Unknown
              component={[Function]}
              disabled={true}
              label="Force Status"
              model="stripDelivery.forceTrafficStatus"
              type="submit"
            />
          </div>
        </styled.div>,
      ],
      "model": "testModel",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <styled.span>
                <Unknown
                  disabled={true}
                  modelPath=".status"
                  status="DELIVER"
                />
              </styled.span>
            </styled.div>,
            <Unknown
              component={[Function]}
              maxLength={255}
              model=".comment"
              placeholder="Force status comments"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.span>
                  <Unknown
                    disabled={true}
                    modelPath=".status"
                    status="DELIVER"
                  />
                </styled.span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "ForceStatusButtonContainer-DELIVER",
                "nodeType": "class",
                "props": Object {
                  "children": <Unknown
                    disabled={true}
                    modelPath=".status"
                    status="DELIVER"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "ForceStatusButton-DELIVER",
                  "nodeType": "function",
                  "props": Object {
                    "disabled": true,
                    "modelPath": ".status",
                    "status": "DELIVER",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "maxLength": 255,
              "model": ".comment",
              "placeholder": "Force status comments",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Unknown
              noIcon={true}
              status="DELIVER"
            />,
            <div>
              <Unknown
                component={[Function]}
                disabled={true}
                label="Force Status"
                model="stripDelivery.forceTrafficStatus"
                type="submit"
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "noIcon": true,
              "status": "DELIVER",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <Unknown
                component={[Function]}
                disabled={true}
                label="Force Status"
                model="stripDelivery.forceTrafficStatus"
                type="submit"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "component": [Function],
                "disabled": true,
                "label": "Force Status",
                "model": "stripDelivery.forceTrafficStatus",
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.div>
              <styled.span>
                <Unknown
                  disabled={true}
                  modelPath=".status"
                  status="DELIVER"
                />
              </styled.span>
            </styled.div>
            <Unknown
              component={[Function]}
              maxLength={255}
              model=".comment"
              placeholder="Force status comments"
            />
          </styled.div>,
          <styled.div>
            <Unknown
              noIcon={true}
              status="DELIVER"
            />
            <div>
              <Unknown
                component={[Function]}
                disabled={true}
                label="Force Status"
                model="stripDelivery.forceTrafficStatus"
                type="submit"
              />
            </div>
          </styled.div>,
        ],
        "model": "testModel",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <styled.span>
                  <Unknown
                    disabled={true}
                    modelPath=".status"
                    status="DELIVER"
                  />
                </styled.span>
              </styled.div>,
              <Unknown
                component={[Function]}
                maxLength={255}
                model=".comment"
                placeholder="Force status comments"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.span>
                    <Unknown
                      disabled={true}
                      modelPath=".status"
                      status="DELIVER"
                    />
                  </styled.span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "ForceStatusButtonContainer-DELIVER",
                  "nodeType": "class",
                  "props": Object {
                    "children": <Unknown
                      disabled={true}
                      modelPath=".status"
                      status="DELIVER"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": "ForceStatusButton-DELIVER",
                    "nodeType": "function",
                    "props": Object {
                      "disabled": true,
                      "modelPath": ".status",
                      "status": "DELIVER",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "component": [Function],
                "maxLength": 255,
                "model": ".comment",
                "placeholder": "Force status comments",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Unknown
                noIcon={true}
                status="DELIVER"
              />,
              <div>
                <Unknown
                  component={[Function]}
                  disabled={true}
                  label="Force Status"
                  model="stripDelivery.forceTrafficStatus"
                  type="submit"
                />
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "noIcon": true,
                "status": "DELIVER",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <Unknown
                  component={[Function]}
                  disabled={true}
                  label="Force Status"
                  model="stripDelivery.forceTrafficStatus"
                  type="submit"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "disabled": true,
                  "label": "Force Status",
                  "model": "stripDelivery.forceTrafficStatus",
                  "type": "submit",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Force status component Renders with default props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    forceTrafficStatus="DELIVER"
    modelPath="testModel"
    patientId="123"
    setDeliveryStatusRequest={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.div>
            <styled.span>
              <Unknown
                disabled={true}
                modelPath=".status"
                status="DELIVER"
              />
            </styled.span>
            <styled.span>
              <Unknown
                disabled={false}
                modelPath=".status"
                status="DO_NOT_DELIVER"
              />
            </styled.span>
          </styled.div>
          <Unknown
            component={[Function]}
            maxLength={255}
            model=".comment"
            placeholder="Force status comments"
          />
        </styled.div>,
        <styled.div>
          <Unknown
            noIcon={true}
            status="DELIVER"
          />
          <div>
            <Unknown
              component={[Function]}
              disabled={false}
              label="Force Status"
              model="stripDelivery.forceTrafficStatus"
              type="submit"
            />
          </div>
        </styled.div>,
      ],
      "model": "testModel",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <styled.span>
                <Unknown
                  disabled={true}
                  modelPath=".status"
                  status="DELIVER"
                />
              </styled.span>
              <styled.span>
                <Unknown
                  disabled={false}
                  modelPath=".status"
                  status="DO_NOT_DELIVER"
                />
              </styled.span>
            </styled.div>,
            <Unknown
              component={[Function]}
              maxLength={255}
              model=".comment"
              placeholder="Force status comments"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.span>
                  <Unknown
                    disabled={true}
                    modelPath=".status"
                    status="DELIVER"
                  />
                </styled.span>,
                <styled.span>
                  <Unknown
                    disabled={false}
                    modelPath=".status"
                    status="DO_NOT_DELIVER"
                  />
                </styled.span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "ForceStatusButtonContainer-DELIVER",
                "nodeType": "class",
                "props": Object {
                  "children": <Unknown
                    disabled={true}
                    modelPath=".status"
                    status="DELIVER"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "ForceStatusButton-DELIVER",
                  "nodeType": "function",
                  "props": Object {
                    "disabled": true,
                    "modelPath": ".status",
                    "status": "DELIVER",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "ForceStatusButtonContainer-DO_NOT_DELIVER",
                "nodeType": "class",
                "props": Object {
                  "children": <Unknown
                    disabled={false}
                    modelPath=".status"
                    status="DO_NOT_DELIVER"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "ForceStatusButton-DO_NOT_DELIVER",
                  "nodeType": "function",
                  "props": Object {
                    "disabled": false,
                    "modelPath": ".status",
                    "status": "DO_NOT_DELIVER",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "maxLength": 255,
              "model": ".comment",
              "placeholder": "Force status comments",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Unknown
              noIcon={true}
              status="DELIVER"
            />,
            <div>
              <Unknown
                component={[Function]}
                disabled={false}
                label="Force Status"
                model="stripDelivery.forceTrafficStatus"
                type="submit"
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "noIcon": true,
              "status": "DELIVER",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <Unknown
                component={[Function]}
                disabled={false}
                label="Force Status"
                model="stripDelivery.forceTrafficStatus"
                type="submit"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "component": [Function],
                "disabled": false,
                "label": "Force Status",
                "model": "stripDelivery.forceTrafficStatus",
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.div>
              <styled.span>
                <Unknown
                  disabled={true}
                  modelPath=".status"
                  status="DELIVER"
                />
              </styled.span>
              <styled.span>
                <Unknown
                  disabled={false}
                  modelPath=".status"
                  status="DO_NOT_DELIVER"
                />
              </styled.span>
            </styled.div>
            <Unknown
              component={[Function]}
              maxLength={255}
              model=".comment"
              placeholder="Force status comments"
            />
          </styled.div>,
          <styled.div>
            <Unknown
              noIcon={true}
              status="DELIVER"
            />
            <div>
              <Unknown
                component={[Function]}
                disabled={false}
                label="Force Status"
                model="stripDelivery.forceTrafficStatus"
                type="submit"
              />
            </div>
          </styled.div>,
        ],
        "model": "testModel",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <styled.span>
                  <Unknown
                    disabled={true}
                    modelPath=".status"
                    status="DELIVER"
                  />
                </styled.span>
                <styled.span>
                  <Unknown
                    disabled={false}
                    modelPath=".status"
                    status="DO_NOT_DELIVER"
                  />
                </styled.span>
              </styled.div>,
              <Unknown
                component={[Function]}
                maxLength={255}
                model=".comment"
                placeholder="Force status comments"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.span>
                    <Unknown
                      disabled={true}
                      modelPath=".status"
                      status="DELIVER"
                    />
                  </styled.span>,
                  <styled.span>
                    <Unknown
                      disabled={false}
                      modelPath=".status"
                      status="DO_NOT_DELIVER"
                    />
                  </styled.span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "ForceStatusButtonContainer-DELIVER",
                  "nodeType": "class",
                  "props": Object {
                    "children": <Unknown
                      disabled={true}
                      modelPath=".status"
                      status="DELIVER"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": "ForceStatusButton-DELIVER",
                    "nodeType": "function",
                    "props": Object {
                      "disabled": true,
                      "modelPath": ".status",
                      "status": "DELIVER",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "ForceStatusButtonContainer-DO_NOT_DELIVER",
                  "nodeType": "class",
                  "props": Object {
                    "children": <Unknown
                      disabled={false}
                      modelPath=".status"
                      status="DO_NOT_DELIVER"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": "ForceStatusButton-DO_NOT_DELIVER",
                    "nodeType": "function",
                    "props": Object {
                      "disabled": false,
                      "modelPath": ".status",
                      "status": "DO_NOT_DELIVER",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "component": [Function],
                "maxLength": 255,
                "model": ".comment",
                "placeholder": "Force status comments",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Unknown
                noIcon={true}
                status="DELIVER"
              />,
              <div>
                <Unknown
                  component={[Function]}
                  disabled={false}
                  label="Force Status"
                  model="stripDelivery.forceTrafficStatus"
                  type="submit"
                />
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "noIcon": true,
                "status": "DELIVER",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <Unknown
                  component={[Function]}
                  disabled={false}
                  label="Force Status"
                  model="stripDelivery.forceTrafficStatus"
                  type="submit"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "disabled": false,
                  "label": "Force Status",
                  "model": "stripDelivery.forceTrafficStatus",
                  "type": "submit",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
